torch/_refs/__init__.py

```diff
@@ -1822,6 +1824,73 @@ def tensor_split(
         return tuple(splits)
 
 
+def hsplit(
+    a: TensorLikeType, indices_or_sections: DimsType
+) -> Tuple[TensorLikeType, ...]:
+    if a.ndim < 1:
+        msg = (
+            "torch.hsplit requires a tensor with at least 1 dimension, but got a tensor with "
+            + str(a.ndim)
+            + " dimensions!"
+        )
+        raise RuntimeError(msg)
+    dim = 0 if a.ndim == 1 else 1
+    if isinstance(indices_or_sections, int):
+        split_size = indices_or_sections
+        if not (split_size != 0 and a.shape[dim] % split_size == 0):
+            msg = (
+                "torch.hsplit attempted to split along dimension "
+                + str(dim)
+                + ", but the size of the dimension "
+                + str(a.shape[dim])
+                + " is not divisible by the split_size "
+                + str(split_size)
+                + "!"
+            )
+            raise RuntimeError()
```

Comment:
msg should go here -- @ezyang added a helper function we should start using, too, see: 

https://github.com/pytorch/pytorch/blob/41e2611e6a3f53f549ad27585a58ba7f7c71594a/torch/_prims/utils.py#L1192